src/services/pacs_integration.tsx

```diff
@@ -0,0 +1,111 @@
+        const responseJSON = JSON.parse(responseBody);
+        return responseJSON;
+    }
+
+    /**
+     * Performs a PACS query on the given PatientID and returns an array of DcmImages containing metadata 
+     * of the DICOM files associated with that patient
+     * @param {string | undefined} patientID 
+     */
+    static async queryPatientFiles(patientID: string | undefined): Promise<DcmImage[]> {
+        if (!patientID) return [];
+        
+        let patientData: DcmImage[] = [];
+        try {
+            const rawResponse = await axios.post(process.env.REACT_APP_CHRIS_UI_PFDCM_URL, JSON.stringify({
+                action:"PACSinteract",
+                meta:{
+                    do:"query",
+                    on:{PatientID: patientID},
+                    PACS:"orthanc"
+                }
+            }), {headers: {'Content-Type': 'text/plain'}})
+            const parsedResponse = this.parseResponse(rawResponse.data);
+            const data: PACSMainResponse[] = parsedResponse.query.data;
+            data.forEach(study => {
+                study.series.forEach(series => {
+                    patientData.push({
+                        id: series.uid.value,
+                        creation_date: series.StudyDate.value,
+                        fname: this.basePACSFilePath + series.SeriesInstanceUID.value,
+                        PatientID: series.PatientID.value,
+                        PatientName: series.PatientName.value,
+                        PatientBirthDate: series.PatientBirthDate.value,
+                        PatientAge: series.PatientAge.value,
+                        PatientSex: series.PatientSex.value,
+                        StudyInstanceUID: series.StudyInstanceUID.value,
+                        StudyDescription: series.StudyDescription.value,
+                        SeriesInstanceUID: series.SeriesInstanceUID.value,
+                        SeriesDescription: series.SeriesDescription.value,
+                        Modality: series.Modality.value,
+                        pacs_identifier: 'covidnet'
+                    })
+                });
+            });
+            
+            return patientData;
+        } catch (err) {
+            return Promise.reject(err);
+        }
+    }
+
+    /**
+     * Performs a PACS retrieve request for a DICOM file based on StudyInstanceUID and SeriesInstanceUID, 
+     * returning true if request was successfully sent
+     * @param {string | undefined} StudyInstanceUID 
+     * @param {string | undefined} SeriesInstanceUID 
+     */
+    static async retrievePatientFiles(StudyInstanceUID: string | undefined, SeriesInstanceUID: string | undefined): Promise<boolean> {
+        interface RetrieveResponse {
+            status: boolean;
+            retrieve: {
+                status: string;
+                data: any[];
+                command: string;
+                returncode: number;
+            }
+        }
```

Comment:
If this interface will be used not just in `retrievePatientFiles`, consider moving the interface to a separate location.

Reply:
Created an interface called `PFDCMResponse` for this.
https://github.com/darwinai/covidnet_ui/pull/8/commits/37be49613bfaa69979a6ee3bfd21b1bdb94816f0